### DISCUSSION TITLE
Understanding the EventFilter / QualifiedName

### DISCUSSION THREAD

**Discussion by dev-chris13** [eclipse-milo/milo#1060] [General]
Hi! 

I am currently trying to build a client that collects specific events via an event subscription. I am struggling to understand the string part of the QualifiedName. I am not sure how to map it to things I see the UAExpert.
 
I am instressed in events of the Type AlarmConditionType and I need to know the TransitionTime of the ActiveState.  What would be the QualifiedName for the TransitionTime?
This are the relationships between these elements:
AlarmCondtionType –hasComponent--> ActiveState --hasProperty--> TransitionTime

Another example would be the TransitionTime of the LastTransition of the ShelvingState: 
AlarmCondtionType –hasComponent--> ShelvingState -- hasComponent --> LastTransition --hasProperty--> TransitionTime

Thanks in advance for your help!
Best,
Chris

**Comment by dev-chris13** [comment]
Found the solution. 

If anyone has the same question: 

Use the  BrowseName as string part of a QulifiedName. Provide a list of QulifiedNames as relative path / browsPath to the node of interest (https://reference.opcfoundation.org/v104/Core/DataTypes/SimpleAttributeOperand/)

Thanks for giving me a space to think "loud".

**Reply by kevinherron** [reply]
@dev-chris13 glad you figured it out. I’ve been away on vacation the last week and just now checking my email etc…

**Reply by dev-chris13** [reply]
Hi Kevin! Thanks for getting back to me! Hope you had a great vacation.

I still have one question: is it possible to read the other elements of value, e.g., Source timestamp?

**Reply by kevinherron** [reply]
If an event has a "Value" property, and you subscribed to it, you would receive a DataValue that has those components in it (value, StatusCode, server/source timestamps if available). You wouldn't ask for these components independently.

**Reply by dev-chris13** [reply]
I checked by comparing the ManagedSubscriptionDataExample and the ManagedSubscriptionEventExample with the ProSysSimulationServer.
When looking at the values in the ManagedSubscriptionDataExample I can find the complete structure of the DataValue and all required information:
![image](https://user-images.githubusercontent.com/77881675/194933434-737d6afe-d96f-42a2-957f-cc10ac89041d.png)
When trying the same for the eventFieldValues in the ManagedSubscriptionEventExample I cannot find this structure which is available according to the UAExpert:
 
![image](https://user-images.githubusercontent.com/77881675/194933449-a5e48b63-6804-4839-8957-035559911fb5.png)
![image](https://user-images.githubusercontent.com/77881675/194933474-55d53bc5-f168-448a-88b6-9d68ffabbe50.png)

Thanks again!

**Reply by kevinherron** [reply]
Event items and data items are two different things. If you want to subscribe to a a VariableNode’s Value attribute you probably need a regular data MonitoredItem, not an event MonitoredItem.
